pkgs/repo_lint_rules/lib/src/dont_import_outside_src_rule.dart

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/error/listener.dart';
+import 'package:custom_lint_builder/custom_lint_builder.dart';
+
+class DontImportOutsideSrcRule extends DartLintRule {
```

Comment:
Could be named (as well as the file name) to `AvoidImportOutsideSrcRule` to reflect the name of the lint.